dialog/src/main/java/roboy/dialog/ConversationManager.java

```diff
@@ -82,63 +82,40 @@ public static void main(String[] args) throws IOException {
         analyzers.add(new ProfanityAnalyzer());
 
 
-        //Roboy mode: Repeat a conversation a few times.
-        if(ConfigManager.ROS_ENABLED) {
+        //I/O specific core behaviour
 
-            Conversation c = createConversation(rosMainNode, analyzers, new Inference(), memory, "local");
+        //telegram needs to initialize the telegram API interface and then go into command mode so it can dynamically spawn, delete and manage incoming conversations with many parallelly interacting users
+        if (ConfigManager.INPUT.contains("telegram")) {
 
-            //Repeat conversation c.
-            for (int numConversations = 0; numConversations < 50; numConversations++) {
-                logger.info("PRESS ENTER TO START THE DIALOG. ROBOY WILL WAIT FOR SOMEONE TO GREET HIM (HI, HELLO)");
-                System.in.read();
+            // initialize telegram bot
+            ApiContextInitializer.init();
+            TelegramBotsApi telegramBotApi = new TelegramBotsApi();
 
-                c.start();
-                try {//Since this is roboy mode and only one conversation happens, we need to wait for it to finish so we don't clog the command line.
-                    logger.info("Waiting for conversation to end.");
-                    c.join();
-                }catch (InterruptedException ie) {
-                    logger.error("ConversationManager has been interrupted: " + ie.getMessage());
-                }
-                //Reset the conversation before rerun.
-                c.resetConversation(new Interlocutor(memory));
+            try {
+                telegramBotApi.registerBot(TelegramCommunicationHandler.getInstance());
+            } catch (TelegramApiException e) {
+                logger.error("Telegram bots api error: ", e);
             }
-        } else {//non-roboy mode
-            if (ConfigManager.INPUT.contains("telegram")) {
 
-                // initialize telegram bot
-                ApiContextInitializer.init();
-                TelegramBotsApi telegramBotApi = new TelegramBotsApi();
+            logger.info("####################################################\n#                SYSTEM LOADED                     #\n####################################################\n");
+            commandMode();
+
+        } else if (ConfigManager.INPUT.contains("cmdline") | ConfigManager.INPUT.contains("cerevoice") | ConfigManager.INPUT.contains("udp") | ConfigManager.INPUT.contains("bing")) {//default single-point of interaction behaviour; start a single conversation, wait for it to end, repeat if necessary; no command mode necessary
```

Comment:
1) i dont really get your choice if bitwise OR operator here. can you explain?
2) we should check if INPUT is "cmdline" etc, not if it contains. since having multiple inputs breaks the state machine

Reply:
@Mireu

Reply:
->1: In plain C it does not make a difference for this usecase. Either way it's very bad style. Sorry.
->2: I changed it into a switch for better readability.